sdk/dotnet/Extensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright 2016-2019, Pulumi Corporation.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Pulumi.Docker
+{
+    internal static class UnwrapExtensions
+    {
+        public static Input<ImmutableArray<T>?> Unwrap<T>(this InputList<T>? items)
+            => items == null ? Output.Create((ImmutableArray<T>?)null) : items.Apply(v => (ImmutableArray<T>?)v);
```

Comment:
feels like this should be returning an `Output<ImmutableArray<T>?>` instead of an `Input<ImmutableArray<T>?>`. 

(Same comment for most of these).

Reply:
Gotcha!

Reply:
I restructured all extensions one more time. The common ones are all expressed in terms of `ToOutput`, `Apply`, and `MapT0/1`. The specific ones  are `Unwrap` that rely on common ones.

The only problem is that I need to preserve null-iness: if an input is nullable, I want the output to contain a nullable, and vice versa. It seems that it's not possible to make nullable and non-nullable overloads of otherwise equal methods, so I had to name them `ToOutputNullable` etc. Not a fan, but I don't see other options yet.